src/python/include/Simplex_tree_interface.h

```diff
@@ -187,6 +196,11 @@ class Simplex_tree_interface : public Simplex_tree<SimplexTreeOptions> {
       collapsed_stree_ptr->insert({std::get<0>(remaining_edge), std::get<1>(remaining_edge)}, std::get<2>(remaining_edge));
     }
     return collapsed_stree_ptr;
+#else
+    // If no Eigen3, return a copy, as it will be deleted in pyx
+    Simplex_tree_interface* collapsed_stree_ptr = new Simplex_tree_interface(*this);
+    return collapsed_stree_ptr;
```

Comment:
If the old one gets destroyed immediately, should we move instead of copy? This feels too quiet, I think we should at least warn, possibly even throw an exception if people try to call this function without Eigen.

Reply:
I was not sure, this was the reason of my comment. I throw an exception on 40e0976